peer/brontide.go

```diff
@@ -2437,19 +2438,14 @@ out:
 			// values, as they currently are always set to the default values
 			// at initial channel creation. Note that the maximum HTLC value
 			// defaults to the cap on the total value of outstanding HTLCs.
-			//
-			// TODO(guggero): We should instead pass in the current
-			// forwarding policy from the funding manager to avoid
-			// needing us to update the link once the channel is
-			// announced to the network with custom user values.
-			fwdMinHtlc := lnChan.FwdMinHtlc()
-			defaultPolicy := p.cfg.RoutingPolicy
+			localCfg := newChan.LocalChanCfg
+			initPolicy := newChanReq.forwardingPolicy
```

Comment:
I don't think we need to use `forwardingPolicy` here. Instead we can do a query using `p.cfg.ChannelDB.GetInitialFwdingPolicy()`.

Reply:
That feels very hacky to be honest. First, the peer package would need to make some assumptions over when and how the data is stored in the DB (given that `p.cfg.ChannelDB.GetInitialFwdingPolicy()` only returns the raw bytes, which we could probably change though). Passing in the initial policies feels way more explicit and IMO is easier to understand in the larger picture of the channel funding flow.

Reply:
> the peer package would need to make some assumptions over when and how the data is stored in the DB

hmmm...what assumptions do we need to make here?

> Passing in the initial policies feels way more explicit and IMO is easier to understand in the larger picture of the channel funding flow.

Guess I'm on the other side, thinking even passing `*channeldb.OpenChannel` needs to be changed by passing a channel ID or any identifier, then let `brontide` query db to get the current channel state.  And `brontide` needs to be slowly refactored to care only about sending/receiving messages, and doesn't need to know the details of the channel.

Reply:
Okay, I'll take another look at this some time next week.